internal/receiver/smartagentreceiver/receiver.go

```diff
@@ -135,5 +145,63 @@ func (r *Receiver) createMonitor(monitorType string) (interface{}, error) {
 		output.AddExtraDimension(k, v)
 	}
 
+	collectdLock.Lock()
+	defer collectdLock.Unlock()
+	// Note, that this receiver has to configure main collectd even for collectd/custom monitor,
+	// despite the fact that, that monitor stands up its own instance of collectd to prevent this
+	// panic "Main collectd instance should not be accessed before being configured".
+	if r.config.monitorConfig.MonitorConfigCore().IsCollectdBased() && !configuredCollectd {
+		r.setUpCollectdConfig(extensions)
+		if err := collectd.ConfigureMainCollectd(r.getCollectdConfig()); err != nil {
+			return nil, err
+		}
+		configuredCollectd = true
+	}
+
 	return monitor, nil
 }
+
+func (r *Receiver) setUpCollectdConfig(extensions map[configmodels.Extension]component.ServiceExtension) {
+	f := smartagentextension.NewFactory()
+	defaultCfg := f.CreateDefaultConfig().(*smartagentextension.Config)
+	r.config.collectdConfig = defaultCfg.CollectdConfig
```

Comment:
Add comment that we fallback to the default if extension is not found. (That's what we do, right?)